android/app/src/main/kotlin/com/chicagoroboto/features/main/MainActivity.kt

```diff
@@ -18,19 +19,17 @@ import com.chicagoroboto.features.speakerdetail.SpeakerDetailActivity
 import com.chicagoroboto.features.speakerdetail.SpeakerNavigator
 import com.chicagoroboto.features.speakerlist.SpeakerListView
 import com.chicagoroboto.model.Session
-import kotlinx.android.synthetic.main.activity_main.content
-import kotlinx.android.synthetic.main.activity_main.drawer_layout
-import kotlinx.android.synthetic.main.activity_main.nav_view
-import kotlinx.android.synthetic.main.activity_main.toolbar
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity(), SessionNavigator, SpeakerNavigator, NavigationView.OnNavigationItemSelectedListener {
-    lateinit var component: MainComponent
+
+    private val component: MainComponent by lazy {
```

Comment:
Curious, what's the benefit to this change?  I'd expect it could potentially be called outside of the lifecycle and could lead to some confusing stack traces, whereas the original approach would just say component is null, and you can look at the call site.

Reply:
The primary reason is to gain immutabilty with the `val` type.
The Stacktrace would be similar if this was still `lateinit var`, because we'd still be defering the initialization.
The Stacktrace would still point to this activity as the root cause.

Reply:
You need to disable the thread safety of the `lazy` otherwise you pay for a double-check lock.

Reply:
Cool, I didn't realize it this was part of backing property.
I pushed a commit to make the thread mode as none.